src/main/java/echoquery/sql/Querier.java

```diff
@@ -4,36 +4,33 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
-import java.sql.DriverManager;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import echoquery.sql.model.ColumnValue;
-import echoquery.utils.EchoQueryCredentials;
+import com.facebook.presto.sql.tree.Query;
+
+import echoquery.sql.formatter.SqlFormatter;
 
 public class Querier {
 
-  private static final Logger log =
-      LoggerFactory.getLogger(Querier.class);
+  private static final Logger log = LoggerFactory.getLogger(Querier.class);
 
   private Connection conn;
   private SchemaInferrer inferrer;
 
-  private Querier() {
-    this.conn = SingletonConnection.getInstance();
+  public Querier(Connection conn) {
+    this.conn = conn;
     this.inferrer = new SchemaInferrer(conn);
   }
 
-  public int count(String table) throws SQLException {
+  public int execute(Query query) throws SQLException {
```

Comment:
who calls this method? I'm a little confused about this class in general. Why does it have an inferrer?